specification/recoveryservicesdatareplication/resource-manager/Microsoft.DataReplication/preview/2021-02-16-preview/recoveryservicesdatareplication.json

```diff
@@ -0,0 +1,7002 @@
+        },
+        "type": {
+          "description": "Gets or sets the type of the resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "systemData": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/SystemDataModel"
+            }
+          ],
+          "readOnly": true
+        },
+        "tags": {
+          "description": "Gets or sets the resource tags.",
+          "maxLength": 15,
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "DraModelCollection": {
+      "description": "Dra model collection.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Gets or sets the list of Dras.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DraModel"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "description": "Gets or sets the value of next link.",
+          "type": "string"
+        }
+      }
+    },
+    "DraModelCustomProperties": {
+      "description": "Dra model custom properties.",
+      "required": [
+        "instanceType"
+      ],
+      "type": "object",
+      "properties": {
+        "instanceType": {
+          "description": "Gets or sets the instance type.",
+          "enum": [
+            "None",
+            "VMware",
+            "AVS",
+            "Azure",
+            "AzStackHCI",
+            "HyperVMigrate",
+            "VMwareMigrate"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "FabricType",
+            "modelAsString": true
+          }
+        }
+      },
+      "discriminator": "instanceType"
+    },
+    "DraModelProperties": {
+      "description": "Dra model properties.",
+      "required": [
+        "authenticationIdentity",
+        "customProperties",
+        "machineId",
+        "machineName",
+        "resourceAccessIdentity"
+      ],
+      "type": "object",
+      "properties": {
+        "correlationId": {
+          "description": "Gets or sets the Dra correlation Id.",
+          "type": "string",
+          "readOnly": true
+        },
+        "machineId": {
+          "description": "Gets or sets the machine Id where Dra is running.",
+          "minLength": 1,
+          "type": "string"
+        },
+        "machineName": {
+          "description": "Gets or sets the machine name where Dra is running.",
+          "minLength": 1,
+          "type": "string"
+        },
+        "authenticationIdentity": {
+          "$ref": "#/definitions/IdentityModel"
+        },
+        "resourceAccessIdentity": {
+          "$ref": "#/definitions/IdentityModel"
+        },
+        "isResponsive": {
+          "description": "Gets or sets a value indicating whether Dra is responsive.",
+          "type": "boolean",
```

Comment:
ARM recommends string enum over boolean for many reasons. Consider something like activation: { Enabled, Disabled } (or similar with better names). Ref: https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean#common-issues-found-in-review.

Reply:
Please consider converting to enums for future extensibility wherever you see the need.